src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -75,8 +74,9 @@ export default function withAnnotations<P extends object>(
             }
         };
 
-        handleOnAnnotator = (annotator: Annotator): void => {
+        handleAnnotator = (annotator: Annotator): void => {
             this.annotator = annotator;
+            this.annotator.addListener('annotatorevent', this.handleAnnotatorEvent);
```

Comment:
You can add listeners for the various events directly rather than listening for `annotatorevent`:

```
this.annotator.addListener('annotations_active_change', this. handleActiveChange)
this.annotator.addListener('annotations_create', this.handleAnnotationsCreate)
```

Also, what happens when the preview instance is destroyed? Should we call removeEventListener in the handlePreviewDestroy method?

Reply:
Preview calls removeAllListeners on the annotator instance

Reply:
That seems like kind of an indirect thing to rely upon. Should we explicitly remove our handlers here, as well, for completeness and safety's sake?

Reply:
Sure, I brought it up earlier and thought it was what we were going to rely on, but I'll add it here 👍

Reply:
Yeah Preview does call `removeAllListeners`, but it seems like from the box-annotations sdk perspective, we should just clean up what we setup instead of relying on a client to clean up for us